src/gui/uithememanager.h

```diff
@@ -36,6 +36,16 @@
 #include <QObject>
 #include <QString>
 
+class UIThemeSource
+{
+public:
+    virtual ~UIThemeSource() = default;
+
+    virtual QByteArray readStyleSheet() const = 0;
+    virtual QByteArray readConfig() const = 0;
```

Comment:
Sorry I wasn't specific enough in the last discussion. Now the `read*` functions shouldn't be `const`. Such as `QIODevice::readAll()` isn't `const`.

`iconPath(...) const` is ok.

Reply:
well `read*` shouldn't modify logical state since it's an verb but whatever

Reply:
> `QIODevice::readAll()` isn't `const`. I/O operations is seldom const qualified.

Perhaps simply because they change something like Cursor, the position of which may be part of the QIODevice state.

Anyway, I don't want to delay it any longer. These are purely utilitarian classes used in a single place. In addition, the real "state" of UIThemeSource isn't so clear. Although I am still convinced that none of its methods are intended to change its state and any of them would look correctly applicable to constant object.